Mathlib/Algebra/Group/Action/Units.lean

```diff
@@ -122,6 +122,34 @@ instance isScalarTower'_left [Group G] [Monoid M] [MulAction G M] [SMul M α] [S
 example [Monoid M] [Monoid N] [MulAction M N] [SMulCommClass M N N] [IsScalarTower M N N] :
     MulAction Mˣ Nˣ := Units.mulAction'
 
+section MulDistribMulAction
+variable {M N : Type*} [Monoid M] [Monoid N] [MulDistribMulAction M N]
+
+/-- Note this instance has different defeqs than `Units.mulAction'`, but doesn't create a diamond
+with it in non-degenerate situations. Indeed, to get a diamond on `MulDistribMulAction G Mˣ`,
+we would need both instances to fire. But `Units.mulAction'` assumes `SMulCommClass G M M`,
+i.e. `∀ (g : G) (m₁ m₂ : M), g • (m₁ * m₂) = m₁ * g • m₂`), while
+`Units.instMulDistribMulActionRight` assumes `MulDistribMulAction G M`,
+i.e. `∀ (g : G) (m₁ m₂ : M), g • (m₁ * m₂) = g • m₁ * g • m₂`.
+In particular, if `M` is cancellative, then we obtain `∀ (g : G) (m : M), g • m = m`,
+i.e. the action is trivial!
+
+This instance however does create a (propeq) diamond for `MulDistribMulAction (ConjAct Mˣ) Mˣ` with
+`ConjAct.unitsMulDistribMulAction` and `ConjAct.instMulDistribMulAction`. Indeed, if we go down
+one way then `u • v := ⟨ofConjAct u * v * ofConjAct u⁻¹, ofConjAct u * v⁻¹ * ofConjAct u⁻¹, _, _⟩`,
+while the other way is
+`u • v := ⟨ofConjAct u * v * ofConjAct u⁻¹, ofConjAct u * (v⁻¹ * ofConjAct u⁻¹), _, _⟩`. -/
+instance instMulDistribMulActionRight : MulDistribMulAction M Nˣ where
```

Comment:
I am not that sure about having this as a global instance. I don't think `SMul (ConjAct Mˣ) Mˣ` is that uncommon.